src/compiler/cbs/VectorShapeTransformer.cpp

```diff
@@ -23,6 +25,14 @@
 using namespace hipsycl::compiler;
 using namespace llvm;
 
+#if LLVM_VERSION_MAJOR < 13
+#define IS_OPAQUE(pointer) if constexpr (false)
+#elif LLVM_VERSION_MAJOR < 16
+#define IS_OPAQUE(pointer) if (pointer->isOpaquePointerTy())
+#else
+#define IS_OPAQUE(pointer) if constexpr (true)
+#endif
+
```

Comment:
@fodinabor I feel like it would be nice to do something like
```cpp
#if LLVM_VERSION_MAJOR < 13
#define IS_OPAQUE(pointer) constexpr (false)
#elif LLVM_VERSION_MAJOR < 16
#define IS_OPAQUE(pointer) (pointer->isOpaquePointerTy())
#else
#define IS_OPAQUE(pointer) constexpr (true)
#endif
```
so that the macro is queried like
```cpp
if IS_OPAQUE(thingy) {
  return something
}
```